src/VecSim/containers/raw_data_container_interface.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+struct RawDataContainer {
+    enum Status { OK = 0, ID_ALREADY_EXIST, ID_NOT_EXIST, ERR };
```

Comment:
It is a better practice to use scoped enumaration (`enum class`) that doesn't allow implicit conversion to integral types.